src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java

```diff
@@ -0,0 +1,149 @@
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import org.apache.commons.collections.CollectionUtils;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBImport;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@BetaFeature
+@CommandLineProgramProperties(summary = "fix the sample names in a vcf that ran through the GenomicsDBImport when " +
+        "the batch ordering bug was present, this will restore the correct sample names provided it is given the exact sample name mapping / vcf ordering" +
+        "and batch sized that was used in the initial import" +
+        "see https://github.com/broadinstitute/gatk/issues/3682 for more information",
+        oneLineSummary = "fix sample names in a shuffled callset",
+        programGroup = VariantProgramGroup.class,
+        omitFromCommandLine = true)
+public final class FixCallSetSampleOrdering extends VariantWalker {
+    public static final String SKIP_PROMPT_LONG_NAME = "skipPrompt";
+
+    @Argument(fullName = GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME,
+            doc="the same sampleNameMap file which was used to import the callset using GenomicsDBImport",
+            optional = false)
+    public String sampleNameMapPath;
+
+    @Argument(fullName = GenomicsDBImport.BATCHSIZE_ARG_NAME,
+            doc="the exact batch size that was used to import the callset using GenomicsDBImport",
+            minValue = 0,
+            optional = false)
+    public Integer batchSize;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="where to write a reheadered version of the input VCF with the sample names in the correct order",
+            optional = false)
+    public File output;
+
+    @Argument(fullName = SKIP_PROMPT_LONG_NAME,
+            shortName = "Y",
+            doc = "skip the commandline prompt as if you had entered Y",
+            optional = true)
+    public boolean skipCommandLinePrompt = false;
+
+    private VariantContextWriter writer;
+
+    @Override
+    public void onTraversalStart() {
+        assertThatTheyReallyWantToProceed();
+
+        if (batchSize == 0) {
+            throw new UserException("your callset is not affected by the bug if you ran with "+ GenomicsDBImport.BATCHSIZE_ARG_NAME +" 0");
```

Comment:
"if you ran with " -> "if you ran with --"

Reply:
done